src/components/Auth/ResetPassword/ForgotPassForm.tsx

```diff
@@ -0,0 +1,111 @@
+'use client'
+
+import Loader from '@/components/UI/Loader/Loader'
+import Button from '@/components/UI/Buttons/Button/Button'
+import FormInput from '@/components/UI/FormInput/FormInput'
+import { useForm } from 'react-hook-form'
+import { useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { IForgotValues } from '@/types/ChangePassword'
+import { apiForgotPassword } from '@/services/userService'
+import { useErrorHandler } from '@/services/apiError/apiError'
+import { forgotPassSchema } from '@/validation/forgotPassSchema'
+import { useLocalSessionStore } from '@/store/useLocalSessionStore'
+
+
+export default function ForgotPassForm() {
+  const [loading, setLoading] = useState(false)
+  const { errorMessage, handleError } = useErrorHandler()
+  const router = useRouter()
+  const setStoredEmailSent = useLocalSessionStore((state) => state.setStoredEmailSent)
```

Comment:
to reduce the code, perhaps we need to destructurize the states and functions from the storage?